pkg/prometheus/statefulset.go

```diff
@@ -765,22 +754,23 @@ func makeStatefulSetSpec(p monitoringv1.Prometheus, c *operator.Config, shard in
 			bindAddress = "127.0.0.1"
 		}
 
-		thanosArgs := []string{"sidecar",
-			fmt.Sprintf("--prometheus.url=%s://%s:9090%s", prometheusURIScheme, c.LocalHost, path.Clean(webRoutePrefix)),
-			fmt.Sprintf("--grpc-address=%s:10901", bindAddress),
-			fmt.Sprintf("--http-address=%s:10902", bindAddress),
+		thanosContainerArgs := []string{"sidecar"}
```

Comment:
I think that we can get rid of this variable and insert `"sidecar"` when building the final args list (e.g. L878 in this version)